wit/batch.wit

```diff
@@ -1,27 +1,81 @@
 /// A keyvalue interface that provides batch operations.
+/// 
+/// A batch operation is an operation that operates on multiple keys at once.
+/// 
+/// Batch operations are useful for reducing network round-trip time. For example, 
+/// if you want to get the values associated with 100 keys, you can either do 100 get
+/// operations or you can do 1 batch get operation. The batch operation is
+/// faster because it only needs to make 1 network call instead of 100.
+/// 
+/// A batch operation does not guarantee atomicity, meaning that if the batch
+/// operation fails, some of the keys may have been modified and some may not. 
+/// Transactional operations are being worked on and will be added in the future to
+/// provide atomicity.
+/// 
+/// Data consistency in a key value store refers to the gaurantee that once a 
+/// write operation completes, all subsequent read operations will return the
+/// value that was written.
+/// 
+/// The level of consistency in batch operations can vary depending on the 
+/// implementation. This interface does not guarantee strong consistency, meaning
+/// that if a write operation completes, subsequent read operations may not return
+/// the value that was written.
 interface batch {
 	/// A keyvalue interface that provides batch get operations.
 	use types.{bucket, error, key, keys, incoming-value, outgoing-value};
 
 	/// Get the values associated with the keys in the bucket. It returns a list of
-	/// incoming-values that can be consumed to get the values.
+	/// incoming-value that can be consumed to get the value associated with the key.
 	///
-	/// If any of the keys do not exist in the bucket, it returns an error.
-	get-many: func(bucket: bucket, keys: keys) -> result<list<incoming-value>, error>;
+	/// If any of the keys do not exist in the bucket, it returns an None value for
+	/// that key in the list.
+	/// 
+	/// Note that the key-value pairs are guaranteed to be returned in the same order
```

Comment:
```suggestion
	/// Note that the key-value pairs are not guaranteed to be returned in the same order
```

I'm not sure if you intended to state this guarantee. In other places you state the opposite.

Reply:
The `get-many` returns a `list<option<incoming-value>>` as the Ok type. If it's not guaranteed to be returend in the same order, then the user will have a difficult time figuring out which value of `None` is corresponding to which key being returned. Hence I added the ORDERING restriction here. 

Alternatively, we can change the return type to `list<(key, option<incoming-value>)>`, and this will be clear to the caller the key-value pairing relation. Then I am more confortable dropping the ORDERING restriction. 

What do you think?

Reply:
I think that is great. I just wanted to be extra sure that was your intention.